compiler/src/main/scala/org/scalajs/core/compiler/PrepJSInterop.scala

```diff
@@ -1027,6 +1036,27 @@ abstract class PrepJSInterop extends plugins.PluginComponent
       ddef
     }
   }
+
+  private def checkInternalAnnotations(tree: Tree): Unit = {
+    def isPreTyper(annotation: AnnotationInfo) =
+      annotation.symbol == WasPublicBeforeTyperClass && annotation.pos.isDefined
+
+    def isPostTyper(annotation: AnnotationInfo) = {
+      annotation.symbol == ExposedJSMemberAnnot ||
+      annotation.symbol == JSFullNameAnnotation ||
+      annotation.symbol == RawJSTypeAnnot ||
+      annotation.symbol == SJSDefinedAnonymousClassAnnotation
+    }
```

Comment:
I'm ok with the names `isPreTyper` and `isPostTyper`, but please add a comment why you need two of them and the shortcuts you take (i.e. a summary of the discussion we had). In that comment, please mention that `@WasPublicBeforeTyper` could still be inserted by a macro, but we can't reliably check for that with reasonable effort.